### PR TITLE
Add heapsort.c

### PR DESCRIPTION
**Pull request by ayswarya141**
Please include a summary of the change. Please also include relevant motivation and context (if applicable).

- [ ] I have performed a self-review of my code
- [ ] My code follows the style guidelines of this project
- [ ] I have added comments to hard-to-understand areas of my code
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have made corresponding changes to the README.md

### CODE REVIEW

**Comment by siriak** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6320511&bid=20545320)

File: heapsort.c [hunk 1]
```diff
@@ -0,0 +1,61 @@
+#include<stdio.h>
+#include <conio.h>
+void Adjust(int Heap_of_Numbers[],int i)  /*Function to arrange the elements in the heap*/
+{
+int j;
```
Comment (siriak) [inline heapsort.c:5]: Codacy found an issue: [The scope of the variable 'j' can be reduced.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6320511)

File: heapsort.c [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+for(i=Number_of_Elements/2;i>=1;i--)
+Adjust(heap,i);
+}
+int main()
+{
+int heap[30];
+int NumberofElements;
+int i;
+int LastElement;
+int CopyVariable;
```
Comment (siriak) [inline heapsort.c:40]: Codacy found an issue: [The scope of the variable 'CopyVariable' can be reduced.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6320511)

File: heapsort.c [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+Number_of_Elements=heap[0];
+for(i=Number_of_Elements/2;i>=1;i--)
+Adjust(heap,i);
+}
+int main()
+{
+int heap[30];
+int NumberofElements;
+int i;
+int LastElement;
```
Comment (siriak) [inline heapsort.c:39]: Codacy found an issue: [The scope of the variable 'LastElement' can be reduced.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6320511)

File: heapsort.c (hunk 2 above)
Comment (ayswarya141) [reply heapsort.c:40]: for swapping pupose we are using copyvariable

File: heapsort.c (hunk 3 above)
Comment (ayswarya141) [reply heapsort.c:39]: lastelement is used to get the last element of the array.

File: heapsort.c (hunk 1 above)
Comment (ayswarya141) [reply heapsort.c:5]: j is the variable used to jump between the elements in an array

**Comment by siriak** [comment]
This is **C#** repo.

**Pull request closed without merging**